test/e2e/lib/helpers/browser.ts

```diff
@@ -1,126 +1,74 @@
-const validBrowserNames = ['edge', 'safari', 'chrome', 'firefox'] as const
-
-export function getBrowserName(): (typeof validBrowserNames)[number] {
-  const capabilities = browser.capabilities
-
-  // Look for the browser name in capabilities. It should always be there as long as we don't change
-  // the browser capabilities format.
-  if (!('browserName' in capabilities) || typeof capabilities.browserName !== 'string') {
-    throw new Error("Can't get browser name (no browser name)")
-  }
-  let browserName = capabilities.browserName.toLowerCase()
-  if (browserName === 'msedge') {
-    browserName = 'edge'
-  } else if (browserName === 'chrome-headless-shell') {
-    browserName = 'chrome'
-  }
-  if (!includes(validBrowserNames, browserName)) {
-    throw new Error(`Can't get browser name (invalid browser name ${browserName})`)
-  }
-
-  return browserName
+export interface BrowserLog {
+  level: 'log' | 'debug' | 'info' | 'error' | 'warning'
+  message: string
+  source: string
+  timestamp: number
 }
 
-export function getPlatformName(): (typeof validPlatformNames)[number] {
-  const capabilities = browser.capabilities
-
-  let platformName: string
-  if ('bstack:options' in capabilities && capabilities['bstack:options']) {
-    // Look for the platform name in browserstack options. It might not be always there, for example
-    // when we run the test locally. This should be adjusted when we are changing the browser
-    // capabilities format.
-    platformName = (capabilities['bstack:options'] as any).os
-  } else {
-    // The test is run locally, use the local os name
-    platformName = os.type()
-  }
+export class BrowserLogsManager {
+  private logs: BrowserLog[] = []
 
-  platformName = platformName.toLowerCase()
-  if (/^(mac ?os|os ?x|mac ?os ?x|darwin)$/.test(platformName)) {
-    platformName = 'macos'
-  } else if (platformName === 'windows_nt') {
-    platformName = 'windows'
-  }
-  if (!includes(validPlatformNames, platformName)) {
-    throw new Error(`Can't get platform name (invalid platform name ${platformName})`)
+  add(log: BrowserLog) {
+    this.logs.push(log)
   }
 
-  return platformName
-}
-
-function includes<T>(list: readonly T[], item: unknown): item is T {
-  return list.includes(item as any)
-}
+  get() {
+    const filteredLogs = this.logs.filter((log) => !log.message.includes('Ignoring unsupported entryTypes: '))
 
-interface BrowserLog {
-  level: string
-  message: string
-  source: string
-  timestamp: number
-}
+    if (filteredLogs.length !== this.logs.length) {
+      // FIXME: fix this at the perfomance observer level as it is visible to customers
+      // It used to pass before because it was only happening in Firefox but wdio io did not support console logs for FF
+      test.info().annotations.push({
+        type: 'dd_tags[test.fixme]',
+        description: 'Unnexpected Console log message: "Ignoring unsupported entryTypes: *"',
+      })
```

Comment:
This add the property `@test.fixme` in CI visibility, this is very useful to create dashboards